R/modules.R

```diff
@@ -196,16 +196,16 @@ module <- function(label = "module",
       "\n - `...` server_args elements will be passed to the module named argument or to the `...`"
     )
   }
+
   if ("datasets" %in% server_formals) {
-    warning(
+    stop(
```

Comment:
I think this needs to be discussed. I know it is one of the acceptance criteria but I'm not sure if we really want this now.

Reply:
Thanks for raising this.

I'm not too keen on adding extra work for us if we end up needing to support our internal end users due to this change.

Let’s hold off on hard-deprecating `datasets` in `module(server)` for now and proceed with the rest.

@llrs-roche @m7pr any concern?

Reply:
Ok, I'll revert back to a warning. 

Perhaps as part of the https://github.com/insightsengineering/nestdevs-tasks/issues/86 we could  decide a policy on the life cycle of functions and arguments? It could incorporate how we search and find internal (looking at internal Roche's code repositories?, asking someone?)  and external repositories (searching github?, asking other programmers from other companies?) that could be affected. Similar to how the tidyverse sends PR to affected packages before sending their new version to CRAN.